src/Util/FileHasher.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Util;
+
+use Rector\Core\Exception\ShouldNotHappenException;
+
+final class FileHasher {
+    /**
+     * @param string[] $files
+     */
+    public function hashFiles(array $files): string
+    {
+        $configHash = '';
+        foreach ($files as $file) {
+            $hash = hash_file('crc32b', $file);
```

Comment:
supported since PHP5.2.6 - https://3v4l.org/2jvJ8

Reply:
I think md4 or xxh128 when exists can be used for consistency, ref https://github.com/rectorphp/rector-src/pull/3508/files#diff-fe54c83c0449653c57f79459ef63e2e4c847e1bad68461add56de99d7db89e54

Reply:
ok, I have centralized all hashing in a single util

Reply:
changing the hashing lead to perf regression. had to revert to the previous state. I would leave it like that for now